build-support/bin/native/bootstrap_rust.sh

```diff
@@ -26,12 +47,11 @@ function bootstrap_rust() {
   )
 
   # Control a pants-specific rust toolchain.
-  if [[ ! -x "${RUSTUP}" ]]
-  then
+  if [[ ! -x "${RUSTUP}" ]]; then
     log "A pants owned rustup installation could not be found, installing via the instructions at" \
         "https://www.rustup.rs ..."
     local -r rustup_tmp=$(mktemp -t pants.rustup.XXXXXX)
-    curl https://sh.rustup.rs -sSf > ${rustup_tmp}
+    curl_rustup_init_script_while_maybe_downgrading > ${rustup_tmp}
     # NB: rustup installs itself into CARGO_HOME, but fetches toolchains into RUSTUP_HOME.
     sh ${rustup_tmp} -y --no-modify-path --default-toolchain none 1>&2
```

Comment:
Should this line still execute if the downgrade URL gets curled? I think the file formats are different - bash script vs. binary.

Reply:
Correct! Will fix.

Reply:
What I've done here is to just `chmod +x` the file in both cases, then execute it directly, as the normal rustup script has a `#!` line and should be able to be executed directly as well.